src/components/panel/Editor.tsx

```diff
@@ -343,6 +394,42 @@ export default function Editor({
 
   const toggleShowOriginal = useCallback(() => setShowOriginal((prev: boolean) => !prev), [setShowOriginal]);
 
+  // Handle double-click zoom cycling (same logic as spacebar)
+  const handleDoubleClick = useCallback(() => {
+    if (isCropping || isMasking || isAiEditing) return;
+    
+    const currentDisplaySize = {
+      width: imageRenderSize.width * transformState.scale,
+      height: imageRenderSize.height * transformState.scale
+    };
+    
+    const currentPercent = originalSize && originalSize.width > 0 && currentDisplaySize.width > 0 
+      ? Math.round((currentDisplaySize.width / originalSize.width) * 100)
+      : 100;
+      
+    let fitPercent = 100;
+    if (originalSize && originalSize.width > 0 && originalSize.height > 0 && baseRenderSize && baseRenderSize.width > 0 && baseRenderSize.height > 0) {
+      const originalAspect = originalSize.width / originalSize.height;
+      const baseAspect = baseRenderSize.width / baseRenderSize.height;
+      
+      if (originalAspect > baseAspect) {
+        fitPercent = Math.round((baseRenderSize.width / originalSize.width) * 100);
+      } else {
+        fitPercent = Math.round((baseRenderSize.height / originalSize.height) * 100);
+      }
+    }
+    
+    if (onZoomChange) {
+      if (Math.abs(currentPercent - fitPercent) < 5) {
+        onZoomChange(1.0);
+      } else if (Math.abs(currentPercent - 100) < 5) {
+        onZoomChange(2.0);
+      } else {
+        onZoomChange('fit-to-window');
```

Comment:
If you really need to give `onZoomChange()` multiple values, add more parameters to the function. For example instead `'fit-to-window'` could be used with `onZoomChange(zoomValue: number, fitToWindow: boolean)` where the boolean being true, the zoom value is ignored when called with `onZoomChange(0, true)`.